src/map.cpp

```diff
@@ -5366,6 +5371,39 @@ bool map::accessible_furniture( const tripoint &f, const tripoint &t, const int
     return ( f == t || clear_path( f, t, range, 1, 100 ) );
 }
 
+std::vector<tripoint> map::get_dir_circle( const tripoint &f, const tripoint &t ) const
+{
+    std::vector<tripoint> circle;
+    circle.resize(8);
+
+    const std::vector<tripoint> line = line_to( f, t, 0, 0 );
+    const std::vector<tripoint> spiral = closest_tripoints_first( 1, f );
+    const std::vector<int> pos_index {1,2,4,6,8,7,5,3};
```

Comment:
Suggestion: make it static, and make it a std::array as it's size does not change at runtime.